git_greppers/utils.sh

```
@@ -1,30 +1,300 @@
+  
+  # Handle HTTPS format: https://github.com/owner/repo.git
+  if [[ "$repo_url" =~ ^https?://.+/.+/.+$ ]]; then
+    local repo_path
+    repo_path=$(echo "$repo_url" | sed -E 's|https?://.+/(.+/.+)(\.git)?|\1|')
+    
+    if [[ "$info_type" == "owner" ]]; then
+      echo "$repo_path" | cut -d '/' -f 1
+    elif [[ "$info_type" == "repo" ]]; then
+      echo "$repo_path" | cut -d '/' -f 2 | sed 's/\.git$//'
+    else  # full
+      echo "$repo_path" | sed 's/\.git$//'
```

Shellcheck (0.10.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)